task_sdk/tests/execution_time/test_task_runner.py

```diff
@@ -78,42 +113,48 @@ def test_parse(test_dags_dir: Path):
     assert isinstance(ti.task.dag, DAG)
 
 
-def test_run_basic(test_dags_dir: Path, time_machine):
+def test_run_basic(time_machine, mocked_parse_fixture):
     """Test running a basic task."""
     what = StartupDetails(
         ti=TaskInstance(id=uuid7(), task_id="hello", dag_id="super_basic_run", run_id="c", try_number=1),
-        file=str(test_dags_dir / "super_basic_run.py"),
+        file="",
         requests_fd=0,
     )
 
-    ti = parse(what)
-
     instant = timezone.datetime(2024, 12, 3, 10, 0)
     time_machine.move_to(instant, tick=False)
 
     with mock.patch(
         "airflow.sdk.execution_time.task_runner.SUPERVISOR_COMMS", create=True
     ) as mock_supervisor_comms:
+        ti = mocked_parse_fixture(what, "super_basic_run", CustomOperator(task_id="hello"))
         run(ti, log=mock.MagicMock())
 
         mock_supervisor_comms.send_request.assert_called_once_with(
             msg=TaskState(state=TerminalTIState.SUCCESS, end_date=instant), log=mock.ANY
         )
 
 
-def test_run_deferred_basic(test_dags_dir: Path, time_machine):
+def test_run_deferred_basic(time_machine, mocked_parse_fixture):
     """Test that a task can transition to a deferred state."""
-    what = StartupDetails(
-        ti=TaskInstance(
-            id=uuid7(), task_id="async", dag_id="super_basic_deferred_run", run_id="c", try_number=1
-        ),
-        file=str(test_dags_dir / "super_basic_deferred_run.py"),
```

Comment:
We should remove that file now too

Reply:
Actually we cannot remove this one yet because it is being used in supervisor tests: `test_supervise_handles_deferred_task`. I mentioned this in the PR desc.